api/src/main/java/marquez/db/migrations/V55_5__BackfillFacets.java

```diff
@@ -0,0 +1,346 @@
+              json_build_object(jsonb_object_keys, event -> 'job' -> 'facets' -> jsonb_object_keys)::jsonb as facet
+          FROM %s le, runs r, jsonb_object_keys(coalesce(event -> 'job' -> 'facets', '{}'::jsonb))
+          WHERE r.uuid = le.run_uuid
+      """;
+  private static final String CREATE_JOB_FACETS_VIEW =
+      "CREATE OR REPLACE VIEW job_facets_view AS"
+          + String.format(JOB_FACETS_DEFINITION, "lineage_events");
+
+  private static final String BACK_FILL_FACETS_SQL =
+      """
+          WITH events_chunk AS (
+              SELECT e.* FROM lineage_events e
+              WHERE
+                  COALESCE(e.created_at, e.event_time) < :createdAt
+                  OR (COALESCE(e.created_at, e.event_time) = :createdAt AND e.run_uuid < :runUuid)
+              ORDER BY COALESCE(e.created_at, e.event_time) DESC, e.run_uuid DESC
+              LIMIT :chunkSize
+          ),
+          insert_datasets AS (
+              INSERT INTO dataset_facets
+              """
+          + String.format(DATASET_FACETS_DEFINITION, "events_chunk")
+          + """
+          ),
+          insert_runs AS (
+              INSERT INTO run_facets
+              """
+          + String.format(RUN_FACETS_DEFINITION, "events_chunk")
+          + """
+          ),
+          insert_jobs AS (
+              INSERT INTO job_facets
+              """
+          + String.format(JOB_FACETS_DEFINITION, "events_chunk")
+          + """
+          )
+          INSERT INTO v55_facet_migration_lock
+          SELECT events_chunk.created_at, events_chunk.run_uuid
+          FROM events_chunk
+          ORDER BY
+              COALESCE(events_chunk.created_at, events_chunk.event_time) ASC,
+              events_chunk.run_uuid ASC
+          LIMIT 1
+          RETURNING created_at, run_uuid;
+      """;
+
+  @Setter private Integer chunkSize = null;
+
+  @Setter private boolean triggeredByCommand = false;
+
+  @Setter private Jdbi jdbi;
+
+  public int getChunkSize() {
+    return chunkSize != null ? chunkSize : DEFAULT_CHUNK_SIZE;
+  }
+
+  @Override
+  public MigrationVersion getVersion() {
+    return MigrationVersion.fromVersion("55.5");
+  }
+
+  @Override
+  public String getDescription() {
+    return "BackFillFacets";
+  }
+
+  @Override
+  public Integer getChecksum() {
+    return null;
+  }
+
+  @Override
+  public boolean isUndo() {
+    return false;
+  }
+
+  @Override
+  public boolean isBaselineMigration() {
+    return false;
+  }
+
+  @Override
+  public boolean canExecuteInTransaction() {
+    return false;
+  }
+
+  @Override
+  public void migrate(Context context) throws Exception {
+    if (context != null) {
+      jdbi = Jdbi.create(context.getConnection());
+    }
+
+    /**
+     * Workaround to register uuid_generate_v4 function to generate uuids. gen_random_uuid() is
+     * available since Postgres 13
+     */
+    jdbi.withHandle(h -> h.createCall("CREATE EXTENSION IF NOT EXISTS \"uuid-ossp\"").invoke());
```

Comment:
It might be possible that the migration might not be applied by a superuser? Can we the uuid using `java.util.UUID`, then inject the value in the inserts?

Reply:
That's a hard one. So, we don't want Java to touch the data. It's good for performance when everything is done in SQLs and no data transfers between backend DB and Java application are required. 

Adding this extension is a workaround for PostgreSQL 12, that we still support. If we stopped supporting PostgreSQL 12, we could use built-in uuid function (https://www.postgresql.org/docs/current/functions-uuid.html) available since version 13. 

Otherwise we can add extra condition here to include extra extension only for PostgreSQL 12. Anyway, I think PostgreSQL 12 users will need a superuser.

Reply:
Ok, that's reasonable. Mind making a note in the migration doc on the superuser requirement?